Neos.Flow/Classes/Mvc/ActionResponse.php

```diff
@@ -264,4 +265,52 @@ public function mergeIntoComponentContext(ComponentContext $componentContext): C
 
         return $componentContext;
     }
+
+    /**
+     * Note this is a special use case method that will apply the internal properties (Content-Type, StatusCode, Location, Set-Cookie and Content)
+     * to the given PSR-7 Response and return a modified response. This is used to merge the ActionResponse properties into a possible HttpResponse
+     * created in a View (see ActionController::renderView()) because those would be overwritten otherwise. Note that any component parameters will
+     * still run through the component chain and will not be propagated here.
+     *
+     * WARNING: Should this ActionResponse contain body content it would replace any content in the given HttpReponse.
+     *
+     * @param ResponseInterface $httpResponse
+     * @return ResponseInterface
+     * @internal
+     */
+    public function applyToHttpResponse(ResponseInterface $httpResponse): ResponseInterface
+    {
+        if ($this->statusCode !== 200) {
+            $httpResponse = $httpResponse
+                ->withStatus($this->statusCode);
+        }
+
+        if ($this->hasContent()) {
+            $httpResponse = $httpResponse->withBody($this->content);
+        }
+
+        if ($this->contentType) {
```

Comment:
CGL nitpick
```suggestion
        if ($this->contentType !== null) {
```

Reply:
Copied from above, but sure, we can make it nicer here than above ;)